x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/SpanFlyout/StickySpanProperties.tsx

```diff
@@ -28,13 +29,30 @@ function getSpanLabel(type: string) {
           defaultMessage: 'Navigation timing'
         }
       );
+    case 'mysql':
+      return 'MySQL';
+    case 'query':
+      return i18n.translate(
+        'xpack.apm.transactionDetails.spanFlyout.spanType.action.query',
+        {
+          defaultMessage: 'Query'
+        }
+      );
     default:
       return type;
   }
 }
 
-function getPrimaryType(type: string) {
-  return first(type.split('.'));
+function getSpanTypes(span: Span) {
+  const { type, subtype, action } = span.span;
+
+  const [primaryType, subtypeFromType, actionFromType] = type.split('.');
+
+  return {
+    type: primaryType,
+    subtype: subtype || subtypeFromType,
+    action: action || actionFromType
+  };
```

Comment:
We don't have to show these for 6.x data:
https://github.com/elastic/kibana/issues/26247#issuecomment-454852925

But if it's easy I guess it can't hurt.